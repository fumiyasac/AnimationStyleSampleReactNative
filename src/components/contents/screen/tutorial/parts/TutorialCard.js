/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  View,
  StyleSheet
} from 'react-native';

// react-native-router-fluxのインポート宣言(Actionを使用)
import { Actions } from 'react-native-router-flux';

// チュートリアル用のアニメーションライブラリ用のコンポーネントのインポート宣言
// ※ ライブラリ名: react-native-app-intro-slider
// https://github.com/Jacse/react-native-app-intro-slider
import AppIntroSlider from 'react-native-app-intro-slider';

export default class TutorialCard extends React.Component {

  // MARK: - Functions

  // チュートリアル表示で「終了」ボタンを押した際の処理
  _onPressOnboadingEndButton = () => {
    Actions.AuthContents();
  };

  // MARK: - Rendering Components

  render() {
    const { tutorialContentsData } = this.props;
    return (
      <View style={styles.background}>
        <AppIntroSlider style={styles.appTutorial}
          slides={tutorialContentsData}
          nextLabel={"次へ"}
          doneLabel={"終了"}
          onDone={ () => { this._onPressOnboadingEndButton() } }
          />
      </View>
    );
  };
}

// MARK: - Component Styles

const styles = StyleSheet.create({
  background: {
    flex: 1
  },
  appTutorial: {
    flex: 1
  }
});