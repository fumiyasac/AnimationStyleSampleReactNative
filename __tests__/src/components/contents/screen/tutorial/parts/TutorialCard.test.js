import 'react-native';
import React from 'react';
import TutorialCard from '../../../../../../../src/components/contents/screen/tutorial/parts/TutorialCard';

import renderer from 'react-test-renderer';

it('TutorialCardコンポーネントが正しく表示されること', () => {

	let stubArray = [
		{
			key: 'tutorial1',
			title: '勉強記録をしっかり記録',
			titleStyle: {
				color: '#ffffff',
				fontWeight: 'bold'
			},
			image: require('../../../../../../../src/components/contents/screen/tutorial/assets/tutorial1.jpg'),
			imageStyle: {
				width: 260,
				height: 260,
				borderRadius: 130,
				borderWidth: 1.0,
				borderColor: '#dddddd',
			},
			backgroundColor: '#879fe7',
		},
		{
			key: 'tutorial2',
			title: '無理なく＆無駄なく',
			titleStyle: {
				color: '#ffffff',
				fontWeight: 'bold'
			},
			image: require('../../../../../../../src/components/contents/screen/tutorial/assets/tutorial2.jpg'),
			imageStyle: {
				width: 260,
				height: 260,
				borderRadius: 130,
				borderWidth: 1.0,
				borderColor: '#dddddd',
			},
			backgroundColor: '#879fe7',
		},
		{
			key: 'tutorial3',
			title: '書籍や参考書を活用',
			titleStyle: {
				color: '#ffffff',
				fontWeight: 'bold'
			},
			image: require('../../../../../../../src/components/contents/screen/tutorial/assets/tutorial3.jpg'),
			imageStyle: {
				width: 260,
				height: 260,
				borderRadius: 130,
				borderWidth: 1.0,
				borderColor: '#dddddd',
			},
			backgroundColor: '#879fe7',
		}
	];

  const tree = renderer.create(
		<TutorialCard tutorialContentsData={stubArray} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});