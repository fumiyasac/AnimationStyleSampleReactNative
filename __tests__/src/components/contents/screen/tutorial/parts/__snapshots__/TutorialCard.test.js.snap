// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialCardコンポーネントが正しく表示されること 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      bounces={false}
      data={
        Array [
          Object {
            "backgroundColor": "#879fe7",
            "image": Object {
              "testUri": "../../../src/components/contents/screen/tutorial/assets/tutorial1.jpg",
            },
            "imageStyle": Object {
              "borderColor": "#dddddd",
              "borderRadius": 130,
              "borderWidth": 1,
              "height": 260,
              "width": 260,
            },
            "key": "tutorial1",
            "title": "勉強記録をしっかり記録",
            "titleStyle": Object {
              "color": "#ffffff",
              "fontWeight": "bold",
            },
          },
          Object {
            "backgroundColor": "#879fe7",
            "image": Object {
              "testUri": "../../../src/components/contents/screen/tutorial/assets/tutorial2.jpg",
            },
            "imageStyle": Object {
              "borderColor": "#dddddd",
              "borderRadius": 130,
              "borderWidth": 1,
              "height": 260,
              "width": 260,
            },
            "key": "tutorial2",
            "title": "無理なく＆無駄なく",
            "titleStyle": Object {
              "color": "#ffffff",
              "fontWeight": "bold",
            },
          },
          Object {
            "backgroundColor": "#879fe7",
            "image": Object {
              "testUri": "../../../src/components/contents/screen/tutorial/assets/tutorial3.jpg",
            },
            "imageStyle": Object {
              "borderColor": "#dddddd",
              "borderRadius": 130,
              "borderWidth": 1,
              "height": 260,
              "width": 260,
            },
            "key": "tutorial3",
            "title": "書籍や参考書を活用",
            "titleStyle": Object {
              "color": "#ffffff",
              "fontWeight": "bold",
            },
          },
        ]
      }
      disableVirtualization={false}
      extraData={750}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      invertStickyHeaders={undefined}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flex": 1,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-around",
                },
                Object {
                  "backgroundColor": "#879fe7",
                  "height": 1334,
                  "paddingBottom": 64,
                  "paddingTop": 20,
                  "width": 750,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 26,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              勉強記録をしっかり記録
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../src/components/contents/screen/tutorial/assets/tutorial1.jpg",
                }
              }
              style={
                Object {
                  "borderColor": "#dddddd",
                  "borderRadius": 130,
                  "borderWidth": 1,
                  "height": 260,
                  "width": 260,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-around",
                },
                Object {
                  "backgroundColor": "#879fe7",
                  "height": 1334,
                  "paddingBottom": 64,
                  "paddingTop": 20,
                  "width": 750,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 26,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              無理なく＆無駄なく
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../src/components/contents/screen/tutorial/assets/tutorial2.jpg",
                }
              }
              style={
                Object {
                  "borderColor": "#dddddd",
                  "borderRadius": 130,
                  "borderWidth": 1,
                  "height": 260,
                  "width": 260,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-around",
                },
                Object {
                  "backgroundColor": "#879fe7",
                  "height": 1334,
                  "paddingBottom": 64,
                  "paddingTop": 20,
                  "width": 750,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 26,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              書籍や参考書を活用
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../src/components/contents/screen/tutorial/assets/tutorial3.jpg",
                }
              }
              style={
                Object {
                  "borderColor": "#dddddd",
                  "borderRadius": 130,
                  "borderWidth": 1,
                  "height": 260,
                  "width": 260,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "bottom": 16,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 16,
            "justifyContent": "center",
            "margin": 16,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, .9)",
              },
              Object {
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 4,
                "width": 10,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, .2)",
              },
              Object {
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 4,
                "width": 10,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, .2)",
              },
              Object {
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 4,
                "width": 10,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 16,
                }
              }
            >
              次へ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
